### PR TITLE
chore: upgrade react-native and react-native-reanimated

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Includes many of the changes specified in the [react-native-upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.71.17&to=0.72.15), some additional fixes to support Flipper, and alterations to usages of `react-native-reanimated` that used now unsupported v1 methods and properties. 

#### Dependency Updates

Updates to the following dependencies:
- `react-native-reanimated@3.15.0`
- `react-native@0.72.15`
- `@react-navigation/material-top-tabs@6.6.14` - updated due to reanimated v3.

Adds the following dependencies:
- `react-native-pager-view@6.3.4` need for `@react-navigation/material-top-tabs` upgrade.

### Test plan

- [x] General app testing locally on iOS
- [x] General app testing locally on Android
- [x] Tested QR Code QR Scanner locally on iOS
- [x] Tested QR Code QR Scanner locally on Android
- [x] E2E tests updated on Android
- [x] Unit Tests updated

### Related issues

- Fixes ACT-1258
- Fixes ACT-1274

### Backwards compatibility

Yes

### Network scalability

N/A

### CODE REVIEW

File: e2e/src/QRScanner.spec.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { quickOnboarding, waitForElementId } from './utils/utils'
 
 // Re-enable for android when react-native-reanimated is updated to v3
```
Comment (bakoushin) [inline e2e/src/QRScanner.spec.js:4]: do we still need this comment?

File: src/components/__snapshots__/FeeIcon.test.tsx.snap [hunk 2]
```diff
@@ -54,9 +54,9 @@ exports[`ExchangeFeeIcon renders correctly 1`] = `
   >
     <svg
       fill="none"
-      height={12}
```
Comment (bakoushin) [inline src/components/__snapshots__/FeeIcon.test.tsx.snap:57]: i wonder why this changed?

**Review by bakoushin — APPROVED** [review]
LGTM! 🚀

File: e2e/src/QRScanner.spec.js (hunk 1 above)
Comment (MuckT) [reply e2e/src/QRScanner.spec.js]: Removed in 9c9b201a436e4deaeb4d5d3cd9d313db799661b8!

File: src/components/__snapshots__/FeeIcon.test.tsx.snap (hunk 2 above)
Comment (MuckT) [reply src/components/__snapshots__/FeeIcon.test.tsx.snap:57]: It appears to be the default sizing that changes I checked a few icons and they looked the same.

**Pull request merged**